packages/cli/src/oclif/hooks/getAppRegistrationFieldChoices.js

```diff
@@ -1,13 +1,30 @@
 const { callAPI } = require('../../utils/api');
+const { BASE_ENDPOINT, IS_TESTING } = require('../../constants');
 
 module.exports = async function (options) {
   // We only need to run this for the register command
   if (!options || !options.id || options.id !== 'register') {
     return null;
   }
 
+  // Mock API call for register field choices when testing
```

Comment:
I'm assuming you are having hard doing this mock within the test cases. Right?

Reply:
@kola-er Correct. I think this is due to us [packing the CLI package](https://github.com/zapier/zapier-platform/blob/4af40fd0b3b7958bc96799095d6fa247b54d07ec/packages/cli/src/smoke-tests/smoke-tests.js#L89) in our `before` block of our smoke tests. For some reason, `nock` isn't able to intercept the http call. Maybe this is because the test cases are [running the `register` command externally](https://github.com/zapier/zapier-platform/blob/4af40fd0b3b7958bc96799095d6fa247b54d07ec/packages/cli/src/smoke-tests/smoke-tests.js#L210) so the API call is outside of `nock`'s scope/access?

My workaround was to implement this mock API call directly into this `getAppRegistrationFieldChoices` hook with the following side effects:
* mock API call isn't present in the test cases (it also isn't obvious, so I should add a comment if we continue with this workaround)
* fixture needs to be [included in `packages/cli/src/oclif` directory](https://github.com/zapier/zapier-platform/blob/4af40fd0b3b7958bc96799095d6fa247b54d07ec/packages/cli/src/oclif/fixtures/registerFieldChoices.js) (instead of in `smoke-tests` directory where it should be) for npm to pack the CLI up properly
* `nock` had to be moved from `devDependencies` to `dependencies` in package.json (see [commit](https://github.com/zapier/zapier-platform/pull/618/commits/5d483845aba7c9332c6d67b83e592607011096d5))
    * This slows down the pack / build process during testing (since we use the [`--production` flag when running `npm install`](https://github.com/zapier/zapier-platform/blob/4af40fd0b3b7958bc96799095d6fa247b54d07ec/packages/cli/src/smoke-tests/smoke-tests.js#L47))

Do you have any thoughts about this workaround? Or alternative approaches?

Reply:
@eliangcs That makes sense to me. I've moved my tests to a new unit test file for the `register` command in 5825c479848f5aec92d67a4809b692cf7493cbf5.